Makefile

```diff
@@ -34,12 +34,20 @@ clean:
 	git clean -df {chalice,daemons/*}/{chalicelib,domovoilib,vendor}
 	git checkout $$(git status --porcelain {chalice,daemons/*}/.chalice/config.json | awk '{print $$2}')
 
+refresh_all_requirements:
+	@echo -n '' >| requirements.txt
+	@echo -n '' >| requirements-dev.txt
+	@if [ $$(uname -s) == "Darwin" ]; then sleep 1; fi  # this is require because Darwin HFS+ only has second-resolution for timestamps.
+	@touch requirements.txt.in requirements-dev.txt.in
+	@echo 'You should now run `make requirements.txt requirements-dev.txt`'
+
 requirements.txt requirements-dev.txt : %.txt : %.txt.in
 	[ ! -e .requirements-env ] || exit 1
-	echo "# You should not edit this file directly.  Instead, you should edit $<." > $@
 	virtualenv .requirements-env
 	source .requirements-env/bin/activate && \
```

Comment:
Since you're touching this, you can eliminate sourcing `activate` by invoking the pip binary in the venv. As in `.requirements-env/bin/pip`. The the `&&` chaining and the \ would go away.

Reply:
#682 
one thesis per diff. :)